geode-core/src/main/java/org/apache/geode/internal/cache/BucketAdvisor.java

```diff
@@ -1192,6 +1195,7 @@ protected boolean acquiredPrimaryLock() {
           if (br instanceof BucketRegionQueue) { // Shouldn't it be AbstractBucketRegionQueue
             BucketRegionQueue brq = (BucketRegionQueue) br;
             brq.incQueueSize(brq.size());
+            brq.decSecondaryQueueSize(brq.size());
```

Comment:
Why do we always decrement the secondary queue size?  It looks like we don't add it unless it's a secondary from below.

Reply:
If reduency equals to 1, I have a test case to prove its necessary. I will add one test case with no secondary. Then I want to see if it has any problems.